File: output/qwen35-27b-vs-qwen36-27b-q5xl-128k/20260423-113141/128k/workspaces/agentic_ts_restore_bootstrap_patch/qwen35_27b_ud_q4_xl/repeat-1/apps/game-runtime/src/runtime/restore.ts
import type { RuntimeState } from "./types";
import { ThreadBootstrapper } from "./bootstrap";

export type Snapshot = {
  agentIds: string[];
  threadIds: Record<string, string>;
};

export function restoreRuntime(
  snapshot: Snapshot,
  state: RuntimeState,
  bootstrapper: ThreadBootstrapper,
): RuntimeState {
  state.phase = "restoring";
  state.threadIds = { ...snapshot.threadIds };
  for (const agentId of snapshot.agentIds) {
    if (!(agentId in state.threadIds)) {
      state.threadIds[agentId] = bootstrapper.ensureThread(agentId);
    }
  }
  state.phase = "running";
  return state;
}

File: output/nemotron-vs-27b/20260326-093607/agentic/workspaces/agentic_python_restore_bootstrap/nemotron_cascade_2_30b_a3b_iq4_xs/repeat-2/runtime/restore.py
from __future__ import annotations

from .session import Snapshot
from .threads import ThreadBootstrapper
from .types import RuntimeState

def restore(snapshot: Snapshot, state: RuntimeState, bootstrapper: ThreadBootstrapper) -> RuntimeState:
    state.phase = "idle"  # Reset phase to idle before processing
    state.thread_ids = {}
    for agent_id in snapshot.agent_ids:
        if agent_id not in state.thread_ids:
            state.thread_ids[agent_id] = bootstrapper.ensure_thread(agent_id)
    state.phase = "running"  # Only set running after all threads are ensured
    state.connected = True
    return state